meson.build

```diff
@@ -54,7 +54,7 @@ cpp = meson.get_compiler('cpp')
 # Prevent use of void* pointer arithmetic to support MSVC
 add_project_arguments(cc.get_supported_arguments('-Werror=pointer-arith'), language : ['cpp', 'c'])
 
-glib_dep = dependency('glib-2.0', version: '>=2.40')
+glib_dep = dependency('glib-2.0', version: '>=2.52')
```

Comment:
The only active Linux I find that this might impact is Slackware 14, which still provides glib 2.46. No idea if that's a problem; there's one way to find out :)

Reply:
Is Slackware 14 still a thing? Slackware 15 provides 2.70

Reply:
I usually pick the lowest supported RHEL version in this case, which is RHEL 7 according to https://endoflife.date/rhel. Fortunately, that provides GLib 2.56.1, so it's probably reasonable to raise this minimum required version to 2.52.

Note to self: revise these conditional guards after this PR has landed.
https://github.com/libvips/libvips/blob/5482ead701b29cbfd3eeab6e91603d972b3b8850/libvips/foreign/pforeign.h#L40
https://github.com/libvips/libvips/blob/5482ead701b29cbfd3eeab6e91603d972b3b8850/libvips/iofuncs/thread.c#L188
https://github.com/libvips/libvips/blob/5482ead701b29cbfd3eeab6e91603d972b3b8850/libvips/iofuncs/init.c#L496